mesonbuild/compilers.py

```diff
@@ -2075,7 +2075,12 @@ def get_buildtype_linker_args(self, buildtype):
         return gnulike_buildtype_linker_args[buildtype]
 
     def get_linker_allow_undefined(self):
-        return ['-Wl,--warn-unresolved-symbols']
+        if self.clang_type == (CLANG_STANDARD, CLANG_WIN):
+            return ['-Wl,--warn-unresolved-symbols']
+        elif self.clang_type == CLANG_OSX:
+            return ['-Wl,--undefined,warning']
```

Comment:
Actually, even `-undefined,warning` should work. Perhaps the problem is that it should be `-undefined`, not `--undefined`. The latter is GNU ld syntax.